### PR TITLE
Introduce filter chip animation brought from iosched 2018

### PR DESCRIPTION
**Pull request by kafumi**
## Issue
- #52
  - #52 is not completed only with this PR
  - Remaining work: Apply coloring to the chips after design spec is updated. See https://github.com/DroidKaigi/conference-app-2019/issues/52#issuecomment-452365784

## Overview (Required)
- Implemented chip animation of filter chips
  - The animatable chips are implemented by custom view `FilterChip`
  - `FilterChip` is copied from iosched's [`EventFilterView`](https://github.com/google/iosched/blob/d8bc48b20c132b519879eda3a8e9af120519e159/mobile/src/main/java/com/google/samples/apps/iosched/ui/schedule/filters/EventFilterView.kt)
- I've modified following points of `EventFilterView` for this app
  - Changed class name to `FilterChip`
    - This view can be used for chips of any types of filter. It might be better to remove "event" from class name
    - I don't have strong preference on it, though
  - Removed build errors, made tiny improvements, and adjusted resource names (e376fb378078620a148a6085937e02b5417974f3)
  - Added `OnCheckedChangeListener` interface. Also, Fixed to perform toggle animation by click event (39937881f994d6f7bf2320923bdbea51059b3411)

## Notes
- Because of the implementation of `EventFilterView`, the chips now support line wrapping for long texts
  - It's also possible to make chips keep single line
  - If single line is preferable, please suggest it. I will do that

## Links
- [`EventFilterView.kt`](https://github.com/google/iosched/blob/d8bc48b20c132b519879eda3a8e9af120519e159/mobile/src/main/java/com/google/samples/apps/iosched/ui/schedule/filters/EventFilterView.kt)

## Screenshot
Before | After
:--: | :--:
svid_20190109_102844_1 | svid_20190109_102635_1

### CODE REVIEW

**Comment by kafumi** [comment]
Sorry, I'm not sure how to resolve "Test results not found" suggestion from review bot. Could I have advices about it?

**Comment by jmatsu** [comment]
It's displayed when no test result is available even though it's required.
Downloading dependencies failed this time so I triggered re-run the CI.

**Comment by kafumi** [comment]
@jmatsu I got it. Sorry not for checking CI build log... Thank you for your help! :pray:

File: frontendcomponent/androidcomponent/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -19,4 +19,7 @@
     <string name="search_label" translatable="false">Search</string>
     <!-- Setting -->
     <string name="setting_label" translatable="false">Setting</string>
+    <!-- Widgets -->
+    <string name="description_filter_applied">%1s filter applied</string>
```
Comment (jmatsu) [inline frontendcomponent/androidcomponent/src/main/res/values/strings.xml]: `%1s` does not work as expected, even though this works fine in this case. 
`%1$s` is a correct argument format. 

Please check the code below on the REPL.

```
fun main(args: Array<String>) {    
    println("%1s".format("xyz"))  // xyz
    println("%1\$s".format("xyz")) // xyz
    println("%1s %1s".format("xyz", "abc")) // xyz abc
    println("%1\$s %1\$s".format("xyz", "abc")) // xyz xyz
}
```

Comment (kafumi) [reply frontendcomponent/androidcomponent/src/main/res/values/strings.xml]: Fixed at 7aa502d22eb74292200c8cdb0ecd9f7cfcd1bf07.

**Pull request merged**